bigquery/src/test/java/org/apache/iceberg/gcp/bigquery/BigQueryTableOperationsTest.java

```diff
@@ -0,0 +1,274 @@
+    ArgumentCaptor<Table> tableArgumentCaptor = ArgumentCaptor.forClass(Table.class);
+    verify(bigQueryMetaStoreClient, times(1))
+        .update(tableReferenceArgumentCaptor.capture(), tableArgumentCaptor.capture());
+    assertThat(tableReferenceArgumentCaptor.getValue()).isEqualTo(TABLE_REFERENCE);
+    assertThat(tableArgumentCaptor.getValue().getEtag()).isEqualTo("etag");
+  }
+
+  @Test
+  public void testFailWhenEtagMismatch() throws Exception {
+    Table tableWithEtag = createTestTable().setEtag("etag");
+    reset(bigQueryMetaStoreClient);
+    when(bigQueryMetaStoreClient.load(TABLE_REFERENCE)).thenReturn(tableWithEtag, tableWithEtag);
+
+    org.apache.iceberg.Table loadedTable = bigQueryMetastoreCatalog.loadTable(SPARK_TABLE_ID);
+
+    when(bigQueryMetaStoreClient.update(any(), any()))
+        .thenThrow(new ValidationException("error message etag mismatch"));
+    assertThatThrownBy(
+            () -> loadedTable.updateSchema().addColumn("n", Types.IntegerType.get()).commit())
+        .isInstanceOf(CommitFailedException.class)
+        .hasMessageContaining(
+            "Updating table failed due to conflict updates (etag mismatch). Retry the update");
+  }
+
+  @Test
+  public void testFailWhenMetadataLocationDiff() throws Exception {
+    Table tableWithEtag = createTestTable().setEtag("etag");
+    Table tableWithNewMetadata =
+        new Table()
+            .setEtag("etag")
+            .setExternalCatalogTableOptions(
+                new ExternalCatalogTableOptions()
+                    .setParameters(ImmutableMap.of(METADATA_LOCATION_PROP, "a/new/location")));
+
+    reset(bigQueryMetaStoreClient);
+    // Two invocations, for loadTable and commit.
+    when(bigQueryMetaStoreClient.load(TABLE_REFERENCE))
+        .thenReturn(tableWithEtag, tableWithNewMetadata);
+
+    org.apache.iceberg.Table loadedTable = bigQueryMetastoreCatalog.loadTable(SPARK_TABLE_ID);
+
+    when(bigQueryMetaStoreClient.update(any(), any())).thenReturn(tableWithEtag);
+    assertThatThrownBy(
+            () -> loadedTable.updateSchema().addColumn("n", Types.IntegerType.get()).commit())
+        .isInstanceOf(CommitFailedException.class)
+        .hasMessageContaining("is not same as the current table metadata location");
+  }
+
+  @Test
+  public void testCreateTableCommitSucceeds() throws Exception {
+    var testTable = createTestTable();
+    when(bigQueryMetaStoreClient.create(any(Table.class))).thenReturn(testTable);
+    when(bigQueryMetaStoreClient.load(
+            new DatasetReference().setProjectId(GCP_PROJECT).setDatasetId(DATASET_ID)))
+        .thenReturn(
+            new Dataset()
+                .setExternalCatalogDatasetOptions(
+                    new ExternalCatalogDatasetOptions()
+                        .setDefaultStorageLocationUri("build/db_folder")));
+
+    Schema schema = testSchema();
+    bigQueryMetastoreCatalog.createTable(SPARK_TABLE_ID, schema, PartitionSpec.unpartitioned());
```

Comment:
there should be some assertions verifying that the table can be loaded or so. Just having code without an assertion looks weird inside test methods

Reply:
I added assertion part too.